modules/core/src/main/java/org/apache/ignite/IgniteSystemProperties.java

```diff
@@ -753,6 +753,16 @@
      */
     public static final String IGNITE_PREFER_WAL_REBALANCE = "IGNITE_PREFER_WAL_REBALANCE";
 
+    /**
+     * Margin for WAL iterator, that used for historical rebalance on atomic cache.
+     * It is intended for prevent  partition divergence due to reordering in WAL.
+     * <p>
+     * Default is {@code 5}. Iterator starts from 5 updates earlier than expected.
+     *
+     */
+    public static final String WAL_MARGIN_FOR_ATOMIC_CACHE_HISTORICAL_REBALANCE =
```

Comment:
I think it looks just right system property. Because we have no information what this value can by.
But in a test where we suspect influence of reordering we can increase this parameter.

Reply:
Done.